Dockerfile

```diff
@@ -41,9 +49,15 @@ COPY --from=pg-build /home/nonroot/pg_install/v16/include/postgresql/server pg_i
 COPY --from=pg-build /home/nonroot/pg_install/v17/include/postgresql/server pg_install/v17/include/postgresql/server
 COPY --from=pg-build /home/nonroot/pg_install/v16/lib                       pg_install/v16/lib
 COPY --from=pg-build /home/nonroot/pg_install/v17/lib                       pg_install/v17/lib
+COPY --from=plan     /home/nonroot/recipe.json                              recipe.json
+
+ARG ADDITIONAL_RUSTFLAGS=""
+
+RUN set -e \
+    && RUSTFLAGS="-Clinker=clang -Clink-arg=-fuse-ld=mold -Clink-arg=-Wl,--no-rosegment -Cforce-frame-pointers=yes ${ADDITIONAL_RUSTFLAGS}" cargo chef cook --release --recipe-path recipe.json
```

Comment:
could you explain why you enable these two parameters?

> --no-rosegment

this seems (mildly) dangerous to me as it means that there is more potential for rop gadgets. mildly because the rx segment probably contains enough already. But I don't really see a benefit in it either?

> force-frame-pointers=yes

Reply:
I don't know what these are doing tbqh. I just copy-pasted them from below

https://github.com/neondatabase/neon/blob/ec354884ea4adf8381a87c8a6486161ca8054e2d/Dockerfile#L70

My assumption is that for caching to work properly we need to pass the same RUSTFLAGS both to `cargo chef cook` and `cargo build`.

Reply:
fair points, it's a separate issue then.